File: devassist-ai/schemas/prologue.py
from pydantic import BaseModel
from typing import Literal

class FocusArea(BaseModel):
    type: Literal["security", "breaking-change", "high-complexity", "data-integrity", "new-pattern", "architecture", "performance", "testing-gap"]
    severity: Literal["critical", "high", "medium", "info"]
    title: str
    description: str

class KeyChangeSummary(BaseModel):
    summary: str
    description: str

class Complexity(BaseModel):
    level: Literal["low", "medium", "high", "very-high"]
    reasoning: str

class PrologueOut(BaseModel):
    motivation: str | None = None
    outcome: str | None = None
    diagram: str | None = None
    key_changes: list[KeyChangeSummary] = []
    focus_areas: list[FocusArea] = []
    complexity: Complexity

File: devassist-ai/frontend/src/lib/api/prologue.ts
export interface PrologueOut {
  motivation?: string;
  outcome?: string;
  diagram?: string;
  focus_areas: Array<{
    type: string;
    severity: string;
    title: string;
    description: string;
  }>;
  complexity: {
    level: string;
    reasoning: string;
  };
}

export async function 
fetchPrologue(reviewId: number): Promise<PrologueOut> {
  const res = await fetch(`/api/v3/reviews/${reviewId}/prologue`);
  if (res.status === 404) return null as any;
  if (!res.ok) throw new Error("Failed to fetch prologue");
  return res.json();
}
